socorro/lib/util.py

```diff
@@ -2,219 +2,147 @@
+        self.messages = []
 
-# utilities
+    def log(self, *args, **kwargs):
+        message = self.create_log_message(*args)
+        self.messages.append(message)
 
-def emptyFilter(x):
-  return (x, None)[x==""]
+    def getMessages(self):
+        log = '\n'.join(self.messages)
+        self.messages = []
+        return log
 
 
-def limitStringOrNone(aString, maxLength):
-  try:
-    return aString[:maxLength]
-  except TypeError:
-    return None
+# logging routines
 
+def setup_logging_handlers(logger, config):
+    stderrLog = logging.StreamHandler()
+    stderrLog.setLevel(config.stderrErrorLoggingLevel)
+    stderrLogFormatter = logging.Formatter(config.stderrLineFormatString)
+    stderrLog.setFormatter(stderrLogFormatter)
+    logger.addHandler(stderrLog)
 
-def lookupLimitedStringOrNone(aDict, aKey, maxLength):
-  try:
-    return limitStringOrNone(aDict[aKey], maxLength)
-  except KeyError:
-    return None
+    syslog = logging.handlers.SysLogHandler(facility=config.syslogFacilityString)
+    syslog.setLevel(config.syslogErrorLoggingLevel)
+    syslogFormatter = logging.Formatter(config.syslogLineFormatString)
+    syslog.setFormatter(syslogFormatter)
+    logger.addHandler(syslog)
 
 
-def lookupStringOrEmptyString(aDict, aKey):
-  try:
-    return aDict[aKey]
-  except KeyError:
-    return ''
+def echo_config(logger, config):
+    logger.info("current configuration:")
+    for value in str(config).split('\n'):
+        logger.info('%s', value)
 
 
-def backoffSecondsGenerator():
-  seconds = [10, 30, 60, 120, 300]
-  for x in seconds:
-    yield x
-  while True:
-    yield seconds[-1]
+logging_report_lock = threading.RLock()
 
 
-class DotDict(dict):
-  __getattr__= dict.__getitem__
-  __setattr__= dict.__setitem__
-  __delattr__= dict.__delitem__
-
-
-class CachingIterator(object):
-  def __init__(self, anIterator):
-    self.theIterator = anIterator
-    self.cache = []
-    self.secondaryLimitedSizeCache = collections.deque()
-    self.secondaryCacheMaximumSize = 11
-    self.secondaryCacheSize = 0
-    self.useSecondary = False
-
-  def close(self):
-    self.theIterator.close()
-
-  def __iter__(self):
-    #try:  #to be used in Python 2.5 or greater
-      for x in self.theIterator:
-        if self.useSecondary:
-          if self.secondaryCacheSize == self.secondaryCacheMaximumSize:
-            self.secondaryLimitedSizeCache.popleft()
-            self.secondaryLimitedSizeCache.append(x)
-          else:
-            self.secondaryLimitedSizeCache.append(x)
-            self.secondaryCacheSize += 1
-        else:
-          self.cache.append(x)
+def report_exception_and_continue(logger=FakeLogger(), loggingLevel=logging.ERROR,
+                                  ignoreFunction=None, showTraceback=True):
+    try:
+        exc_type, exc, tb = sys.exc_info()
+        if ignoreFunction and ignoreFunction(exc_type, exc, tb):
+            return
+        if exc_type in (KeyboardInterrupt, SystemExit):
+            raise
```

Comment:
This is the only thing I changed that's behaviorally different. This used to be:

```
raise exception
```

The problem with that is that it's raising a *new* exception with a new traceback and I think this should be re-raising the exception that was raised.

I think this was a bug. Not a big one because those two particular exceptions are "the user has killed this process intentionally", but still, not great.